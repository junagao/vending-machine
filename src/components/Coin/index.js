export { default as CoinItem } from './CoinItem';
export { default as CoinList } from './CoinList';
export { default as CoinSlot } from './CoinSlot';
export { default as CollectRefund } from './CollectRefund';
export { default as DisplayDepositedAmount } from './DisplayDepositedAmount';
export { default as RefillCoins } from './RefillCoins';
export { default as Wallet } from './Wallet';
