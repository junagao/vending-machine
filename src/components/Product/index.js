export { default as CollectProduct } from './CollectProduct';
export { default as ProductList } from './ProductList';
export { default as ProductItem } from './ProductItem';
export { default as RefillProductStock } from './RefillProductStock';
