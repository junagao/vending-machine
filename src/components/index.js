export { default as CoinSlot } from 'components/CoinSlot';
export { default as CollectProduct } from 'components/CollectProduct';
export { default as VendingMachine } from 'components/VendingMachine';
export { default as VendingMachineScreen } from 'components/VendingMachineScreen';
export { default as Wallet } from 'components/Wallet';
export { default as ProductList } from 'components/ProductList';
export { default as Header } from 'components/Header';
export { default as CollectRefund } from 'components/CollectRefund';
