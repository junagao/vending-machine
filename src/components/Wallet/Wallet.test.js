import React from 'react';
import { shallow } from 'enzyme';
import { Wallet } from 'components';

describe('Wallet', () => {
  let wrapper;
  const coinsMock = [
    {
      id: 'coin1',
      name: '2 euros',
      value: 2,
      userQuantity: 10,
      machineQuantity: 2,
    },
    {
      id: 'coin2',
      name: '1 euro',
      value: 1,
      userQuantity: 5,
      machineQuantity: 3,
    },
  ];

  const walletAmountMock = 21;

  beforeEach(() => {
    wrapper = shallow(
      <Wallet
        walletAmount={walletAmountMock}
        coins={coinsMock}
        onDragStart={jest.fn()}
        onDragEnd={jest.fn()}
        onDragOver={jest.fn()}
        isDragging
      />,
    );
  });

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one Card element', () => {
    expect(wrapper.find('Card').length).toEqual(1);
  });

  it('should render one CardTitle element with the title', () => {
    expect(wrapper.find('CardTitle').length).toEqual(1);
    expect(
      wrapper
        .find('CardTitle')
        .children()
        .at(0)
        .text(),
    ).toEqual(`My wallet amount: `);
    expect(
      wrapper
        .find('CardTitle')
        .children()
        .at(1)
        .text(),
    ).toEqual(`21`);
    expect(
      wrapper
        .find('CardTitle')
        .children()
        .at(2)
        .text(),
    ).toEqual(` â‚¬`);
  });

  it('should render two CoinList element', () => {
    expect(wrapper.find('CoinList').length).toEqual(1);
  });
});
