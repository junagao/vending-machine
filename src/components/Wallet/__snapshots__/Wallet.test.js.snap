// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet should render correctly given the required props 1`] = `
<Card>
  <CardTitle>
    My wallet amount: 
    21
     â‚¬
  </CardTitle>
  <CoinList
    coins={
      Array [
        Object {
          "id": "coin1",
          "machineQuantity": 2,
          "name": "2 euros",
          "userQuantity": 10,
          "value": 2,
        },
        Object {
          "id": "coin2",
          "machineQuantity": 3,
          "name": "1 euro",
          "userQuantity": 5,
          "value": 1,
        },
      ]
    }
    isDragging={true}
    onDragEnd={[MockFunction]}
    onDragOver={[MockFunction]}
    onDragStart={[MockFunction]}
  />
</Card>
`;
